_posts/2022-11-01-setup-a-ci-pipeline-using-github-actions-and-docker.md

```diff
@@ -0,0 +1,188 @@
+
+# Project creation
+
+## Board and IDE
+
+To give you an example of how to set up a continuous integration pipeline in a firmware project, I will use the [Silicon Labs Thunderboard Sense Board](https://www.silabs.com/development-tools/thunderboard/thunderboard-sense-two-kit?tab=overview) to run a simple project on [Simplicity Studio IDE](https://www.silabs.com/developers/simplicity-studio) that we will use to build and perform static analysis.
+
+![Thunderboard](/img/setup-a-ci-pipeline-using-github-actions-and-docker/0-thunderboard.png)
+
+## Create the project and compile it on the Local Machine
+
+If you have the Thunderboard Sense board, you can connect it to the PC and check if the Simplicity Studio detects it.
+
+![Board Selection](/img/setup-a-ci-pipeline-using-github-actions-and-docker/1-board-selection.png)
+
+If you do not have the board, you can add it to my products tab and create the project anyway. It is not impeditive to continue. You can follow the steps in the next images.
+
+![Add new board](/img/setup-a-ci-pipeline-using-github-actions-and-docker/2-add-new-board.png)
+
+![New board](/img/setup-a-ci-pipeline-using-github-actions-and-docker/3-add-board.png)
+
+After you connect or add the board, go to the Launcher perspective on Simplicity Studio, click on the board, and next click on Create New Project button.
+
+![New project](/img/setup-a-ci-pipeline-using-github-actions-and-docker/4-new-project.png)
+
+Select the Empty C Project and Finish!
+
+![Empty project](/img/setup-a-ci-pipeline-using-github-actions-and-docker/5-empty-c.png)
+
+To generate the project Makefile, open the empty.slcp file and do the following steps.
+
+![Generators](/img/setup-a-ci-pipeline-using-github-actions-and-docker/6-generators.png)
+
+Select the GCC Makefile option on Change Project Generators, save, and Force Generation.
+
+![Makefile Generator](/img/setup-a-ci-pipeline-using-github-actions-and-docker/7-gcc-makefile-gen.png)
+
+After that, the empty.Makefile and empty.project.mak files will be created in the project root directory.
+
+![Makefiles](/img/setup-a-ci-pipeline-using-github-actions-and-docker/8-makefile.png)
+
+Now you can build the project in the terminal since Simplicity Studio installed the toolchain.
+
+If you are running it on Linux, you can go directly to the terminal, but if you are using Windows, you will need to install [Cygwin](https://www.cygwin.com/) and execute the command.
+
+```bash
+make -f empty.Makefile
+```
+
+You, at this point, can compile the project at your local machine using the terminal.
+
+![Makefiles](/img/setup-a-ci-pipeline-using-github-actions-and-docker/9-compile.png)
+
+# Test the compilation on a Docker Container
+
+We will use a docker image with Simplicity Studio to build our project on a container. This will allow us to configure GitHub Actions to use the docker image to do the same.
+
```

Comment:
Ah, Simplicity Studio requires login :confused: always a bit tedious when vendors do that for free toolchains.

I'd still like to have the Dockerfile detailed here; providing a pre-built one is great if someone wants to try it, but the heart of the article is about **setting up** such a system, rather than the actual result (eg, if someone's not using Simplicity Studio, having the Dockerfile reference is much more valuable than the pre-built image, IMO).

Reply:
Oh, understood. I will do it.